### PR TITLE
type delete when delete detector; enhance confirm modal

### PR DESCRIPTION
**Pull request by ylwu-amzn**
*Issue #, if available:*

*Description of changes:*
 
* Ask user to type `delete` before delete detector
* Enhance confirm modal to support saving edit features

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/DetectorDetail/containers/DetectorDetail.tsx [hunk 1]
```diff
@@ -225,20 +268,28 @@ export const DetectorDetail = (props: DetectorDetailProps) => {
...
                   <EuiHealth color="subdued">
                     {detector.featureAttributes &&
                     detector.featureAttributes.length
                       ? detector.disabledTime
                         ? `Stopped at ${moment(detector.disabledTime).format(
-                            'MM/DD/YY h:mm a'
+                            'MM/DD/YY h:mm A'
                           )}`
                         : 'Detector is not started'
                       : 'Feature required to start the detector'}
```
Comment (ohltyler) [inline public/pages/DetectorDetail/containers/DetectorDetail.tsx:295]: can use the `DETECTOR_STATE_COLORS` enum here for consistency w/ detector list page. can call `stateToColorMap(detector.curState)` to retrieve the color

Comment (ylwu-amzn) [reply public/pages/DetectorDetail/containers/DetectorDetail.tsx:295]: Sure, seems this needs some testing and tune. Will create an issue to track this. After we merged code, will work on this.

Comment (ohltyler) [reply public/pages/DetectorDetail/containers/DetectorDetail.tsx:295]: Yeah, sounds good

Comment (ylwu-amzn) [reply public/pages/DetectorDetail/containers/DetectorDetail.tsx:295]: Track with this issue: https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/issues/56

File: public/pages/DetectorDetail/containers/DetectorDetail.tsx [hunk 2]
```diff
@@ -117,6 +126,14 @@ export const DetectorDetail = (props: DetectorDetailProps) => {
     }
   }, [detector]);
 
+  const handleSwitchToConfigurationTab = useCallback(() => {
+    setDetecorDetailModel({
```
Comment (ohltyler) [inline public/pages/DetectorDetail/containers/DetectorDetail.tsx:130]: small typo: rename to `setDetectorDetailModel`

Comment (ylwu-amzn) [reply public/pages/DetectorDetail/containers/DetectorDetail.tsx:130]: Sure, will change

**Pull request merged**